core/domain/takeout_service_test.py

```diff
@@ -876,16 +909,109 @@ def test_export_data_for_full_user_trivial_is_correct(self):
             self.USER_ID_1)
         observed_data = user_takeout_object.user_data
         observed_images = user_takeout_object.user_images
+        self.maxDiff = None
         self.assertEqual(expected_user_data, observed_data)
         observed_json = json.dumps(observed_data)
         expected_json = json.dumps(expected_user_data)
         self.assertEqual(json.loads(expected_json), json.loads(observed_json))
         expected_images = []
         self.assertEqual(expected_images, observed_images)
 
+    def test_exports_follow_export_policies(self):
+        """Nontrivial test to ensure that all fields that should be exported
+        per the export policy are exported, and exported in the proper format.
+        """
+        self.set_up_non_trivial()
+
+        # We set up the feedback_thread_model here so that we can easily
+        # access it when computing the expected data later.
+        feedback_thread_model = feedback_models.GeneralFeedbackThreadModel(
+            entity_type=self.THREAD_ENTITY_TYPE,
+            entity_id=self.THREAD_ENTITY_ID,
+            original_author_id=self.USER_ID_1,
+            status=self.THREAD_STATUS,
+            subject=self.THREAD_SUBJECT,
+            has_suggestion=self.THREAD_HAS_SUGGESTION,
+            summary=self.THREAD_SUMMARY,
+            message_count=self.THREAD_MESSAGE_COUNT
+        )
+        feedback_thread_model.put()
+
+        thread_id = feedback_services.create_thread(
+            self.THREAD_ENTITY_TYPE,
+            self.THREAD_ENTITY_ID,
+            self.USER_ID_1,
+            self.THREAD_SUBJECT,
+            self.MESSAGE_TEXT
+        )
+        feedback_services.create_message(
+            thread_id,
+            self.USER_ID_1,
+            self.THREAD_STATUS,
+            self.THREAD_SUBJECT,
+            self.MESSAGE_TEXT
+        )
+
+        # Retrieve all models for export.
+        all_models = [
+            clazz
+            for clazz in self._get_model_classes()
+            if not clazz.__name__ in self.BASE_CLASSES
+        ]
+
+        # Iterate over models and test export policies.
+        for model in all_models:
+            export_method = model.get_export_method()
+            export_policy = model.get_export_policy()
+            exported_props = [
```

Comment:
props --> property_names
prop --> property_name

(here and elsewhere)

Reply:
Done